deploy/cdk/lambda_stack.py

```diff
@@ -157,58 +142,40 @@ def _python_lambda(self, name, directory, env=None, timeout_seconds=900, **kwarg
             **kwargs,
         )
 
-    def _lambda_sg_for_db(self, construct_id, database_vpc):
-        # Security group for db-write lambda
-        lambda_function_security_group = ec2.SecurityGroup(
-            self,
-            f"{construct_id}-lambda-sg",
-            vpc=database_vpc,
-            description="fromCloudOptimizedPipelineLambdas",
-        )
-        lambda_function_security_group.add_egress_rule(
-            ec2.Peer.any_ipv4(),
-            connection=ec2.Port(protocol=ec2.Protocol("ALL"), string_representation=""),
-            description="Allow lambda security group all outbound access",
-        )
-        return lambda_function_security_group
+    def give_permissions(self):
+        internal_bucket = self._bucket(config.VEDA_DATA_BUCKET)
+        internal_bucket.grant_read_write(self.cogify_lambda.role)
 
-    @property
-    def lambdas(self):
-        return self._lambdas
+        external_buckets = [
+            self._bucket(bucket) for bucket in config.VEDA_EXTERNAL_BUCKETS
+        ]
 
-    @property
-    def lambda_sg(self):
-        return self._lambda_sg
+        for bucket in [internal_bucket, *external_buckets]:
+            bucket.grant_read(self.s3_discovery_lambda.role)
+            bucket.grant_read(self.build_stac_lambda.role)
+            if self.data_transfer_lambda:
+                bucket.grant_read(self.data_transfer_lambda.role)
```

Comment:
Is this not updating the bucket policy? It should be fine for the non-NASA deployments, but we won't be able to use it as is in a NASA system. Updating bucket policy is not available to us.

Reply:
I agree that it does look like this is an action that would affect the `bucket` object, [the docs](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_s3.Bucket.html#grantwbrreadidentity-objectskeypattern) validate that it alters the provided principal:

> Grant read permissions for this bucket and it's contents to an IAM principal (Role/Group/User).